src/util/mapHelpersLatLong.js

```diff
@@ -94,134 +122,65 @@ export const getLatLongs = (nodes, visibility, metadata, map, colorBy, geoResolu
   offsets.forEach((OFFSET) => {
     /* count DEMES */
     _.forOwn(aggregatedLocations, (value, key) => {
-      demesAndTransmissions.demes.push({
-        location: key, // Thailand:
-        total: value.length, // 20, this is an array of all demes of a certain type
-        color: averageColors(value),
-        coords: map.latLngToLayerPoint( /* interchange. this is a leaflet method that will tell d3 where to draw. -Note (A) we MAY have to do this every time rather than just once */
-          new L.LatLng(
-            metadata.geo[geoResolution][key].latitude,
-            metadata.geo[geoResolution][key].longitude + OFFSET
-          )
-        )
-      });
+      let lat = geo[geoResolution][key].latitude;
+      let long = geo[geoResolution][key].longitude + OFFSET;
+      if (long > westBound && long < eastBound) {
+        demesAndTransmissions.demes.push({
+          location: key, // Thailand:
+          total: value.length, // 20, this is an array of all demes of a certain type
+          color: averageColors(value),
+          coords: leafletLatLongToLayerPoint(lat, long, map)
```

Comment:
I did this to reduce code duplication. `leafletLatLongToLayerPoint` should be handing back a `new L.LatLng(lat, long)` in the exact same fashion as before, just shelved out into a function.